pkg/ibm/ibmp_helpers.go

```diff
@@ -342,9 +350,38 @@ func (pw IBMPowerDynamicConfig) updateVolume(ctx context.Context, service *core.
 	var vRef models.VolumeReference
 	_, err = service.Request(request, &vRef)
 	if err != nil {
-		log.Error(err, "failed to update PVM volume")
-		return fmt.Errorf("failed to update volume %s: %w", volumeId, err)
+		return fmt.Errorf("failed to update volume %s: %w", volumeID, err)
 	}
-	log.Info("Volume size updated", "volumeId", vRef.VolumeID, "size", vRef.Size)
+	log.Info("Volume size updated", "volumeID", vRef.VolumeID, "size", vRef.Size)
 	return nil
 }
+
+// doesInstanceHaveTaskRun returns whether or not an instance on the pw cloud is associated with an existing
+// TaskRun. Each instance should have a tag with the associated TaskRun's namespace and name. This is compared
+// to existingTaskRuns, which is a map of namespaces to a list of TaskRuns in that namespace, to determine if
+// this instance's TaskRun still exists.
+func (pw IBMPowerDynamicConfig) doesInstanceHaveTaskRun(instance *models.PVMInstance, log logr.Logger, existingTaskRuns map[string][]string) bool {
+	validTaskRunId := false
+
+	// Iterate over this instance's tags
+	for _, tag := range instance.UserTags {
+		err := cloud.ValidateTaskRunId(tag)
+		if err != nil {
+			continue
+		}
+		validTaskRunId = true
+		// Try to find this instance's TaskRun
+		taskRunInfo := strings.Split(tag, ":")
+		taskRunNamespace, taskRunName := taskRunInfo[0], taskRunInfo[1]
+		taskRuns, ok := existingTaskRuns[taskRunNamespace]
+		// Return true if the associated exists in a TaskRun namespace
+		if ok && !slices.Contains(taskRuns, taskRunName) {
+			return true
+		}
+		return false
```

Comment:
This can be simplified 


```suggestion
		return ok && !slices.Contains(taskRuns, taskRunName)
```